text/0089-metrics-views-api.md

```diff
@@ -0,0 +1,264 @@
+
+| System     | Metric Type                                                                                        | OT Metric Instrument | OT Aggregation    |
+| ---------- | -------------------------------------------------------------------------------------------------- | -------------------- | ----------------  |
+| OpenCensus | Count                                                                                              | Counter              | Sum               |
+| OpenCensus | Mean                                                                                               | Counter              | Mean\*            |
+| OpenCensus | Sum                                                                                                | Counter              | Sum               |
+| OpenCensus | Distribution                                                                                       | Measure              | Histogram         |
+| OpenCensus | LastValue                                                                                          | Observer             | LastValue         |
+| Datadog    | [COUNT](https://docs.datadoghq.com/developers/metrics/types/?tab=count#metric-types)               | Counter              | Sum               |
+| Datadog    | [RATE](https://docs.datadoghq.com/developers/metrics/types/?tab=rate#metric-types)                 | Counter              | Rate\*            |
+| Datadog    | [GAUGE](https://docs.datadoghq.com/developers/metrics/types/?tab=gauge#metric-types)               | Observer             | LastValue         |
+| Datadog    | [HISTOGRAM](https://docs.datadoghq.com/developers/metrics/types/?tab=histogram#metric-types)       | Measure              | Histogram         |
+| Datadog    | [DISTRIBUTION](https://docs.datadoghq.com/developers/metrics/types/?tab=distribution#metric-types) | Measure              | Sketch\* or Exact |
+| Prometheus | [Counter](https://prometheus.io/docs/concepts/metric_types/#counter)                               | Counter              | Sum               |
+| Prometheus | [Gauge](https://prometheus.io/docs/concepts/metric_types/#gauge)                                   | Observer             | LastValue         |
+| Prometheus | [Histogram](https://prometheus.io/docs/concepts/metric_types/#histogram)                           | Measure              | Histogram         |
+| Prometheus | [Summary](https://prometheus.io/docs/concepts/metric_types/#summary)                               | Measure              | Sketch\* or Exact |
+
+A well-designed view API should not require extensive configuration for typical use cases.
+It should also be expressive enough to support custom aggregations and exposition formats without requiring APM vendors to write custom SDKs.
+In short, it should make easy things easy and hard things possible.
+
+## Internal details
+
+A view is defined by:
```

Comment:
Do we allow users to create/modify Views in the middle of the program? Or Views must be created at initialization time?
I think the View Config should be provided at the time of MeterProvider construction, before any Meters are created. So that meters know about the which are the required Views, before the 1st metric bind or recording comes.

Reply:
I do not believe we should allow users to do that. There are some [issues](https://github.com/open-telemetry/opentelemetry-python/pull/596#discussion_r411819266) with defining Views after a collection pass has been done, it gets too complicated to deal with those so we should enforce Views to be created and immutable before recording.

Reply:
I need to understand the details a bit more on the issues, but there's tremendous value to being able to modify views on runtime. In a conversation with @c24t a while back, he mentioned that there's a way to configure views across application dynamically for debugging purposes. 

For example, imagine that one of the keys you store is the service name sending you the request. When your application starts seeing a significant increase in requests, you can turn on a view that splits the metrics by service name, and see which service is causing issues. Once resolved, you can turn them off, ensuring that you don't have high cardinality data points that consume significant metric count after the fact.